File: CPE/Exam/231017/23E4_UVA993.py
from collections import *

T = int(input())

for _ in range(T):
	N = int(input())
	if N == 0: # 0的話直接輸出0
		print(0)
		continue
	# 因數分解，檢查有沒有大於10的因數，注意可以合併2^3=8來縮小答案
	cnt = defaultdict(int)
	for p in range(9, 1, -1):
		while (N % p == 0):
			cnt[p] += 1
			N //= p
	if N != 1:
		print(-1)
	else:
		ans = ""
		for p in range(2, 10):
			ans += str(p) * cnt[p]
		print(ans if ans else 1)

File: CPE/Exam/231017/23E4_UVA993.cpp
#include <bits/stdc++.h>
using namespace std;
using LL = long long;
const int INF = 0x3f3f3f3f;
const int MOD = 1e9 + 7;
const int N = 100005;
#define endl '\n'

int main() {
	ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
	int t; LL n;
	cin >> t;
	while (t--) {
		cin >> n;
		if (n == 0) {
			cout << 0 << endl;
			continue;
		}
		int cnt[10] = {0};
		for (int i = 9; i > 1; --i) {
			while (n % i == 0) {
				cnt[i]++;
				n /= i;
			}
		}
		if (n != 1) cout << -1 << endl;
		else {
			string ans = "";
			for (int i = 2; i < 10; ++i) {
				ans += string(cnt[i], i + '0');
			}
			cout << (ans.empty() ? "1" : ans) << endl;
		}
	}
	return 0;
}